lib/elixir/src/elixir_env.erl

```diff
@@ -133,6 +133,8 @@ format_error({unused_var, Name, Overridden}) ->
   case atom_to_list(Name) of
     "_" ++ _ ->
       io_lib:format("unknown compiler variable \"~ts\" (expected one of __MODULE__, __ENV__, __DIR__, __CALLER__, __STACKTRACE__)", [Name]);
+    "&" ++ _ ->
+      io_lib:format("variable \"~ts\" is unused (this might happen when using a capture argument as a pattern)", [Name]);
```

Comment:
What do you think about this message @josevalim ?
- it's probably one of the only case, so the hint might be helpful esp. because `&match?(&1, :value)` might seem valid at first sight
- the same warning gets logged twice, but given this is the first time this issue is reported, I expect it to be uncommon enough so we don't need to add complexity to prevent it? (and potentially fail to warn in a legit case)